lib/site_prism/element_checker.rb

```diff
@@ -3,15 +3,29 @@
 module SitePrism
   module ElementChecker
     def all_there?
-      elements_to_check.all? { |element| send "has_#{element}?" }
+      elements_to_check.all? { |element| present?(element) }
+    end
+
+    def elements_present
+      mapped_items.select { |item_name| present?(item_name) }
     end
 
     private
 
     def elements_to_check
-      elements = self.class.mapped_items
-      elements = elements.select { |el| self.class.expected_items.include?(el) } if self.class.expected_items
-      elements
+      if self.class.expected_items
+        mapped_items.select { |el| self.class.expected_items.include?(el) }
+      else
+        mapped_items
+      end
+    end
+
+    def mapped_items
+      self.class.mapped_items.uniq
+    end
+
+    def present?(element)
```

Comment:
`present?` is already a method added by activesupport on all object as the opposite of `blank?`  ( /lib/active_support/core_ext/object/blank.rb )

I do not think it is a good idea to shadow this public method by a private one.

Reply:
Hi there. Valid point and I think it's worth a change as it's a simple bugfix ticket.

If you have the capacity, feel free to raise a PR. Otherwise I'll get round to it as and when.

Can I ask if you're using a late(r), version of site_prism and capybara. As Ideally I'd just want this hitting the 3.x series (As backporting then just means extra code / gem releases) - Are you happy for it just to hit the 3.x series?

Reply:
I hit the problem on 2.17 but it's ok I have a small `site_prism_patch.rb` loaded by my capybara initializer :)

I'll try to make a PR against  master.

PS : I 'm taking the opportunity of this ticket to say a HUUUUGE thank you to you guys ! I've been using site_prism  since 1.2 and advocating everyone to use it. Adopting it is -  hands down - the best refactoring every single capybara test suite should make.